rero_ils/modules/patrons/api.py

```diff
@@ -190,20 +227,65 @@ def delete(self, force=False, delindex=False):
         super(Patron, self).delete(force, delindex)
         return self
 
+    @classmethod
+    def update_from_profile(cls, profile):
+        """Update the current record with the user profile data.
+
+        :param profile - the rero user profile
+        """
+        # retrieve the user
+        patron = Patron.get_patron_by_user(profile.user)
+        if patron:
+            for field in cls.profile_fields:
+                # date field requires conversion
+                if field == 'birth_date':
+                    patron[field] = getattr(
+                        profile, field).strftime('%Y-%m-%d')
+                else:
+                    patron[field] = getattr(profile, field)
```

Comment:
Suggestion/Idea (optional) :: Is it possible that `cls.profile_fields` return a dict where key are the field name and value could be the function to get the patron.field value ? I will easy to add a new field to synchronize in the future (if needed)

Reply:
Thanks for the suggestion. This would be a good idea, but seems a bit overkill as we need to define a set and get function for each field. An other solution is to create a new children Profile class with a getter and setter. Again it seems a bit overkill for only one field.